File: Additional Sheet (Condition)/N.cpp
#include <iostream>

using namespace std;

int main(){
    long long l1, r1, l2, r2, k;
    cin >> l1 >> r1 >> l2 >> r2 >> k;
    
    if(max(l1, l2) > min(r1, r2)){
        cout << 0;
    }else if(max(l1, l2) <= k && k <= min(r1, r2)){
        cout << min(r1, r2) - max(l1, l2);
    }else{
        cout << min(r1, r2) - max(l1, l2) + 1;
    }
}

File: Additional Sheet (Condition)/N.py
l1, r1, l2, r2, k = map(int, input().split())

start = max(l1, l2)
end = min(r1, r2)

result = 0
if start <= end:
    result = end - start + 1
    if start <= k <= end:
        result -= 1

print(result)
